gpkit/form_program.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+
+from .nomials import Signomial
+from .geometric_program import GeometricProgram
+from .signomial_program import SignomialProgram
+from .nomial_data import sort_and_simplify
+
+from .substitution import substitution
+from .small_scripts import mag
+
+
+def form_program(programType, signomials, subs, verbosity=2):
+    """Generates a program and solves it, sweeping as appropriate.
+
+    Arguments
+    ---------
+    programType : "gp" or "sp"
+
+    signomials : list of Signomials
+        The first Signomial is the cost function.
+
+    subs : dict
+        Substitutions to do before solving.
+
+    verbosity : int (optional)
+        If greater than 0 prints runtime messages.
+        Is decremented by one and then passed to program inits.
+
+    Returns
+    -------
+    program : GP or SP
+    mmaps : Map from initial monomials to substitued and simplified one.
+            See small_scripts.sort_and_simplify for more details.
```

Comment:
since this already returns a tuple, what about having it return cost, constraints, mmaps, and leave out the last-step program creation?

That would eliminate `programType` and `verbosity`, and make this a function that does substitution on a list of Signomials. (move to `substitution.py`?)